### DISCUSSION TITLE
Performance on Apple Silicon

### DISCUSSION THREAD

**Discussion by andrewssdd** [oobabooga/text-generation-webui#1500] [Q&A]
I've been using llama.cpp and experimenting with using a nice GUI like this one.  I use a Macbook M1 for testings.

Tested with the same ggml model:

https://huggingface.co/chharlesonfire/ggml-vicuna-7b-4bit/blob/main/ggml-vicuna-7b-q4_0.bin

Using llama.cpp directly is a lot fast than using the webui. Is it expected or I had the wrong setting? 

I am wondering if this webui calls llama.cpp or has another implementation.

**Comment by panoptic-chaos** [comment]
I'm also seeing a dramatic performance difference between llama.cpp and webui 
Using a 13b model and 8 threads, I see ~0.5 tokens/sec on webui, while llama.cpp is doing 70-85ms/token.

**Comment by JayceEdie** [comment]
I, too, am having the same issue, and any help would be greatly appreciated.

**Comment by merien-io** [comment]
Make sure you have set n-gpu-layers to 1.

**Reply by StuckInSprings** [reply]
I've done this on my end and testing with llama.cpp directly I'm still getting way faster performance. Out of curiosity I tested with another app Faraday and got much faster performance, more in line with what I was seeing with llama.cpp directly. I made sure that metal was being used by oobabooga but still it's the slowest (by a large margin) of the three. I still can't figure out why.
